models/Document.php

```diff
@@ -607,7 +607,7 @@ public function saveIndex($index)
     public function clearDependentCache($additionalTags = [])
     {
         try {
-            $tags = [$this->getCacheTag(), 'document_properties', 'output'];
+            $tags = [$this->getCacheTag(), 'document_properties_'.$this->getId(), 'output'];
```

Comment:
```suggestion
            $tags = [$this->getCacheTag(), 'document_properties_' . $this->getId(), 'output'];
            $children = $this->getChildren();
            foreach ($children as $child){
                $additionalTags[] = 'document_properties_' . $child->getId();
            }
```
might work but for DataObject need some more refactoring

Reply:
But the children can also have children. This is not be enough.

Reply:
🤦🏻‍♂️ oh no, that's right, i get it now why it was "simpler" to clear all properties

Reply:
another case is on deletion: document `abc` has a property that links to document `xyz`; `xyz` got deleted, the cached property of `abc` still contains `xyz` reference

Reply:
Despite method being called `clearDependentCache`, it doesn't go though Dependencies either. and even if it did,  dependencies aren't accurate/clean as well (https://github.com/pimcore/pimcore/pull/15427) 😮‍💨 

I think it's not gonna make it for 10.6, which is EOL next week. 
Let's reconsider this for 11, if it's fine for you